apps/worker/src/app/workflow/usecases/message-matcher/message-matcher.usecase.ts

```diff
@@ -1,525 +1,522 @@
-  IWebhookFilterPart,
-  IRealtimeOnlineFilterPart,
-  IOnlineInLastFilterPart,
-  FILTER_TO_LABEL,
-  FilterPartTypeEnum,
-  ICondition,
-  TimeOperatorEnum,
-  ChannelTypeEnum,
-  IPreviousStepFilterPart,
-  PreviousStepTypeEnum,
-  ExecutionDetailsSourceEnum,
-  ExecutionDetailsStatusEnum,
-} from '@novu/shared';
-import {
-  SubscriberEntity,
-  EnvironmentRepository,
-  SubscriberRepository,
-  StepFilter,
-  ExecutionDetailsRepository,
-  MessageRepository,
-  JobRepository,
-} from '@novu/dal';
-import {
-  DetailEnum,
-  CreateExecutionDetails,
-  CreateExecutionDetailsCommand,
-  buildSubscriberKey,
-  CachedEntity,
-  Instrument,
-  Filter,
-  FilterProcessingDetails,
-  IFilterVariables,
-} from '@novu/application-generic';
-import { EmailEventStatusEnum } from '@novu/stateless';
-
-import { EXCEPTION_MESSAGE_ON_WEBHOOK_FILTER, createHash, PlatformException } from '../../../shared/utils';
-import { MessageMatcherCommand } from './message-matcher.command';
-
-const differenceIn = (currentDate: Date, lastDate: Date, timeOperator: TimeOperatorEnum) => {
-  if (timeOperator === TimeOperatorEnum.MINUTES) {
-    return differenceInMinutes(currentDate, lastDate);
-  }
-
-  if (timeOperator === TimeOperatorEnum.HOURS) {
-    return differenceInHours(currentDate, lastDate);
-  }
-
-  return differenceInDays(currentDate, lastDate);
-};
-
-@Injectable()
-export class MessageMatcher extends Filter {
-  constructor(
-    private subscriberRepository: SubscriberRepository,
-    private createExecutionDetails: CreateExecutionDetails,
-    private environmentRepository: EnvironmentRepository,
-    private executionDetailsRepository: ExecutionDetailsRepository,
-    private messageRepository: MessageRepository,
-    private jobRepository: JobRepository
-  ) {
-    super();
-  }
-
-  public async filter(
-    command: MessageMatcherCommand,
-    variables: IFilterVariables,
-    prefiltering = false
-  ): Promise<{
-    passed: boolean;
-    conditions: ICondition[];
-  }> {
-    const { step } = command;
-    if (!step?.filters || !Array.isArray(step?.filters)) {
-      return {
-        passed: true,
-        conditions: [],
-      };
-    }
-    if (step.filters?.length) {
-      const details: FilterProcessingDetails[] = [];
-
-      const foundFilter = await this.findAsync(step.filters, async (filter) => {
-        const filterProcessingDetails = new FilterProcessingDetails();
-        filterProcessingDetails.addFilter(filter, variables);
-
-        const children = filter.children;
-        const noRules = !children || (Array.isArray(children) && children.length === 0);
-        if (noRules) {
-          return true;
-        }
-
-        const singleRule = !children || (Array.isArray(children) && children.length === 1);
-        if (singleRule) {
-          const result = await this.processFilter(variables, children[0], command, filterProcessingDetails);
-          if (!prefiltering) {
-            await this.createExecutionDetails.execute(
```

Comment:
I can not find this detail somewhere in the new code.

Reply:
Are you referring to the createExecutionDetails execution or `if (singleRule)` block?

Reply:
to the `createExecutionDetails `

Reply:
we will still log the execution detail in `send-message.usecase.ts` if shouldRun.passed is false, and we won't print in queue-next-job.usecase.ts the same as it was before because of the flag. or did u miss something? 

for the moment i remove the responsibility of the createExecutionDetails execution from the filter, do you think we should return it? 

@davidsoderberg

Reply:
But the one of this type is not used anymore?

Reply:
Not sure I understood what you mean here

Reply:
@davidsoderberg i will merge this one for now, please let me know if i missed something here 🙏